### PR TITLE
Add MDAO index on SOURCE_ID as implementation detail of ConfluenceSyncDAO

### PR DESCRIPTION
**Pull request by mdittmer**
This yields substantial improvements in server-side junction DAO performance.

### CODE REVIEW

File: lib/sync_dao.es6.js [hunk 1]
```diff
@@ -97,15 +101,24 @@ foam.CLASS({
 
         // TODO(markdittmer): Clean this up. Need a way to override lookup and
         // requiresCreate.
-        var lookup = this.lookup.bind(this);
+        const lookup = this.lookup.bind(this);
+
+        const indexedMDAO = lookup('foam.dao.MDAO').create({
+          of: this.of,
+        }, this.container.ctx);
+        // Index releaes ID on release/interface junction DAO.
```
Comment (foolip) [inline lib/sync_dao.es6.js]: typo: releaes

File: lib/sync_dao.es6.js [hunk 2]
```diff
@@ -97,15 +101,24 @@ foam.CLASS({
...
+        if (this.of === pkg.VersionedReleaseWebInterfaceJunction) {
```
Comment (arobins) [inline lib/sync_dao.es6.js]: Does the MDAO have any benefits (or drawbacks) if you don't hit this codepath? This feels like an OO antipattern, and the Right Way would usually be to subclass for this case.

File: lib/sync_dao.es6.js [hunk 3]
```diff
@@ -132,7 +145,7 @@ foam.CLASS({
       factory: function() {
         // TODO(markdittmer): Clean this up. Need a way to override lookup and
         // requiresCreate.
-        var lookup = this.lookup.bind(this);
+        const lookup = this.lookup.bind(this);
```
Comment (arobins) [inline lib/sync_dao.es6.js]: Unrelated change?

File: lib/sync_dao.es6.js (hunk 2 above)
Comment (mdittmer) [reply lib/sync_dao.es6.js]: Do you mean implement `VersionedReleaseWebInterfaceJunctionMDAO` with default `of` class and default property index setup?

File: lib/sync_dao.es6.js (hunk 3 above)
Comment (mdittmer) [reply lib/sync_dao.es6.js]: Yeah. I guess.

**Comment by mdittmer** [comment]
@arobins PTAL at response. I agree is-an-instance is a bit hacky, but I want to clarify the alternative.

File: lib/sync_dao.es6.js (hunk 2 above)
Comment (arobins) [reply lib/sync_dao.es6.js]: Something like that
